ssl/ssl_ciph.c

```diff
@@ -995,14 +995,8 @@ static int ssl_cipher_process_rulestr(const char *rule_str,
             ch = *l;
             buf = l;
             buflen = 0;
-#ifndef CHARSET_EBCDIC
-            while (((ch >= 'A') && (ch <= 'Z')) ||
-                   ((ch >= '0') && (ch <= '9')) ||
-                   ((ch >= 'a') && (ch <= 'z')) ||
-                   (ch == '-') || (ch == '.') || (ch == '='))
-#else
-            while (isalnum(ch) || (ch == '-') || (ch == '.') || (ch == '='))
-#endif
+            while (!(ch & ~127) && (isalnum(ch) || ch == '-' || ch == '.'
+                                    || ch == '='))
```

Comment:
The TL;DR response is yes, revert to what it used to do.
I'll assume that the isXXX functions are always operating in their native character sets.
This isn't entirely clear but it makes a sound assumption.